corpus/convert.py

```diff
@@ -12,6 +18,94 @@
 Path = setup_path(__package__)
 
 
+class CorpusReplaceOrthFromReferenceCorpus(Job):
+    """
+    Copies the orth tag from one corpus to another through matching segment names.
+    """
+
+    def __init__(self, bliss_corpus, reference_bliss_corpus):
+        """
+        :param bliss_corpus: Corpus in which the orth tag is to be replaced
+        :param reference_bliss_corpus: Corpus from which the orth tag replacement is taken
+        """
+        self.bliss_corpus = bliss_corpus
+        self.reference_bliss_corpus = reference_bliss_corpus
+
+        self.out_corpus = self.output_path("corpus.xml.gz")
+
+    def tasks(self):
+        yield Task("run", mini_task=True)
+
+    def run(self):
+        orth_c = corpus.Corpus()
+        orth_c.load(tk.uncached_path(self.reference_bliss_corpus))
+
+        orths = {}
+        for r in orth_c.all_recordings():
+            for i in range(len(r.segments)):
+                orth = r.segments[i].orth
+                tag = r.segments[i].fullname()
+                orths[tag] = orth
+
+        c = corpus.Corpus()
+        c.load(tk.uncached_path(self.bliss_corpus))
+
+        for r in c.all_recordings():
+            for i in range(len(r.segments)):
+                tag = r.segments[i].fullname()
+                orth = orths[tag]
```

Comment:
For the `orths[tag]` you should give a more verbose error message in case the key does not exist.

Reply:
Okay like this or is assert overkill / do you want to just catch the Exception and change the error message